src/test/java/io/supertokens/test/bulkimport/ProcessBulkImportUsersCronJobTest.java

```diff
@@ -225,18 +228,55 @@ public void shouldThrowTenantDoesNotExistError() throws Exception {
         BulkImportSQLStorage storage = (BulkImportSQLStorage) StorageLayer.getStorage(main);
         AppIdentifier appIdentifier = new AppIdentifier(null, null);
 
+        // Create user roles before inserting bulk users
+        {
+            UserRoles.createNewRoleOrModifyItsPermissions(main, "role1", null);
+            UserRoles.createNewRoleOrModifyItsPermissions(main, "role2", null);
+        }
+
         List<BulkImportUser> users = generateBulkImportUser(1);
         BulkImport.addUsers(appIdentifier, storage, users);
 
         Thread.sleep(6000);
 
-        List<BulkImportUser> usersAfterProcessing = storage.getBulkImportUsers(appIdentifier, null, null,
+        List<BulkImportUser> usersAfterProcessing = storage.getBulkImportUsers(appIdentifier, 100, null,
+                null, null);
+
+        assertEquals(1, usersAfterProcessing.size());
+        assertEquals(BULK_IMPORT_USER_STATUS.FAILED, usersAfterProcessing.get(0).status);
+        assertEquals(
+                "[Invalid tenantId: t1 for a user role., Invalid tenantId: t1 for a user role., Invalid tenantId: t1 for emailpassword recipe., Invalid tenantId: t1 for thirdparty recipe., Invalid tenantId: t1 for passwordless recipe.]",
+                usersAfterProcessing.get(0).errorMessage);
+    }
+
+    @Test
+    public void shouldThrowTenantHaveDifferentStoragesError() throws Exception {
+        TestingProcess process = startCronProcess();
+        Main main = process.getProcess();
+
+        BulkImportSQLStorage storage = (BulkImportSQLStorage) StorageLayer.getStorage(main);
+        AppIdentifier appIdentifier = new AppIdentifier(null, null);
+
+        // Create user roles before inserting bulk users
+        {
+            UserRoles.createNewRoleOrModifyItsPermissions(main, "role1", null);
+            UserRoles.createNewRoleOrModifyItsPermissions(main, "role2", null);
+        }
+        createTenants(main);
+
+
+        List<BulkImportUser> users = generateBulkImportUser(1, List.of("t1", "t2"), 0);
+        BulkImport.addUsers(appIdentifier, storage, users);
+
+        Thread.sleep(6000);
+
+        List<BulkImportUser> usersAfterProcessing = storage.getBulkImportUsers(appIdentifier, 100, null,
                 null, null);
 
         assertEquals(1, usersAfterProcessing.size());
         assertEquals(BULK_IMPORT_USER_STATUS.FAILED, usersAfterProcessing.get(0).status);
         assertEquals(
-                "Tenant with the following connectionURIDomain, appId and tenantId combination not found: (, public, t1)",
+                "[All tenants for a user must share the same storage for emailpassword recipe., All tenants for a user must share the same storage for thirdparty recipe., All tenants for a user must share the same storage for passwordless recipe.]",
```

Comment:
User won't understand what storage is. The error message should say something like all tenants for a user must use same database.

Reply:
Fixed.